Sources/DatadogObjc/DDURLSessionDelegate+objc.swift

```diff
@@ -8,4 +8,49 @@ import Foundation
 import Datadog
 
 @objc
-public class DDNSURLSessionDelegate: DDURLSessionDelegate {}
+open class DDNSURLSessionDelegate: NSObject, URLSessionTaskDelegate {
+    let swiftDelegate: DDURLSessionDelegate
+
+    @objc
+    override public init() {
+        swiftDelegate = DDURLSessionDelegate()
+    }
+
+    @objc
+    public init(additionalFirstPartyHosts: Set<String>) {
+        swiftDelegate = DDURLSessionDelegate(additionalFirstPartyHosts: additionalFirstPartyHosts)
+    }
+
+    /// NOTE: RUMM-1234
+    /// `DDNSURLSessionDelegate` imitates `DDURLSessionDelegate` class
+    /// it responds to whatever `DDURLSessionDelegate` responds to and
+    /// it forwards every method call to its private `DDURLSessionDelegate` instance.
+    /// That way, `DDNSURLSessionDelegate` inherits `DDURLSessionDelegate` functionality
+    /// without actual inheritance/superclass.
+
+    // swiftlint:disable implicitly_unwrapped_optional
+    override open func forwardingTarget(for aSelector: Selector!) -> Any? {
+        return swiftDelegate
+    }
+
+    override open class func forwardingTarget(for aSelector: Selector!) -> Any? {
+        return DDURLSessionDelegate.self
+    }
+
+    override open func responds(to aSelector: Selector!) -> Bool {
+        return swiftDelegate.responds(to: aSelector)
+    }
+
+    override open class func responds(to aSelector: Selector!) -> Bool {
+        return DDURLSessionDelegate.responds(to: aSelector)
+    }
+    // swiftlint:enable implicitly_unwrapped_optional
+
+    override open func isKind(of aClass: AnyClass) -> Bool {
+        return swiftDelegate.isKind(of: aClass)
+    }
+
+    override open class func isKind(of aClass: AnyClass) -> Bool {
+        return DDURLSessionDelegate.isKind(of: aClass)
+    }
```

Comment:
This changes not only the kind of our delegate, but also its user-defined superclass. This can be seen when defining test for delegate's type appearing at runtime:

<img width="792" alt="Screenshot 2021-04-08 at 11 03 37" src="https://user-images.githubusercontent.com/2358722/113999323-3b0b3180-985a-11eb-967c-0d35530749ea.png">

Changing the runtime identity of user object is dangerous, as it will break any user logic which depends on the delegate kind.